scripts/Utils.py

```diff
@@ -1,13 +1,56 @@
-import requests
 import sys
+import os
+import winreg
+
+import requests
 import time
+import urllib
+
+from zipfile import ZipFile
+
+def GetSystemEnvironmentVariable(name):
+    key = winreg.CreateKey(winreg.HKEY_LOCAL_MACHINE, r"System\CurrentControlSet\Control\Session Manager\Environment")
+    try:
+        return winreg.QueryValueEx(key, name)[0]
+    except:
+        return None
 
-from fake_useragent import UserAgent
+def GetUserEnvironmentVariable(name):
+    key = winreg.CreateKey(winreg.HKEY_CURRENT_USER, r"Environment")
+    try:
+        return winreg.QueryValueEx(key, name)[0]
+    except:
+        return None
 
 def DownloadFile(url, filepath):
+    path = filepath
+    filepath = os.path.abspath(filepath)
+    os.makedirs(os.path.dirname(filepath), exist_ok=True)
+            
+    if (type(url) is list):
+        for url_option in url:
+            print("Downloading", url_option)
+            try:
+                DownloadFile(url_option, filepath)
+                return
+            except urllib.error.URLError as e:
+                print(f"URL Error encountered: {e.reason}. Proceeding with backup...\n\n")
+                os.remove(filepath)
+                pass
+            except urllib.error.HTTPError as e:
+                print(f"HTTP Error  encountered: {e.code}. Proceeding with backup...\n\n")
+                os.remove(filepath)
+                pass
+            except:
+                print(f"Something went wrong. Proceeding with backup...\n\n")
+                os.remove(filepath)
+                pass
+        raise ValueError(f"Failed to download {filepath}")
+    if not(type(url) is str):
+        raise TypeError("Argument 'url' must be of type list or string")
+
     with open(filepath, 'wb') as f:
-        ua = UserAgent()
-        headers = {'User-Agent': ua.chrome}
+        headers = {'User-Agent': "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.97 Safari/537.36"}
         response = requests.get(url, headers=headers, stream=True)
         total = response.headers.get('content-length')
 
```

Comment:
Due to the small file size of license.txt I ran into a small error, which I already fixed in the unzip function, but now also rises in the download function:

1. Due to the small file size, there is an issue as downloaded > total (due to rounding), which can even become 0 in some cases
2. Because of a decent download speed and the [inaccurate timer inside python](https://stackoverflow.com/questions/1938048/high-precision-clock-in-python/21301124#:~:text=time()%20function%20provides%20sub,problems%20due%20to%20process%20interrupts.), I encounter a download speed of 0, which also results in a `ZeroDivisionError` when calculating the avgKBPerSecond.

In order to resolve these issues, could you replace lines 66-69
https://github.com/TheCherno/Hazel/blob/531b9af00c1e5aef037f8475e409790bbcd8855f/scripts/Utils.py#L66-L69
to
```python
                try:
                    done = int(50*downloaded/total) if downloaded < total else 50
                    percentage = (downloaded / total) * 100 if downloaded < total else 100
                except ZeroDivisionError:
                    done = 50
                    percentage = 100
                elapsedTime = time.time() - startTime
                try:
                    avgKBPerSecond = (downloaded / 1024) / elapsedTime
                except ZeroDivisionError:
                    avgKBPerSecond = 0.0
```